operator/v1/types_dns.go

```diff
@@ -376,6 +398,18 @@ const (
 	NetworkResolverType   UpstreamType = "Network"
 )
 
+// +kubebuilder:validation:Enum:=None;ForceTCP;""
+type ProtocolPolicy string
+
+var (
+	// ProtocolPolicyNone specifies no DNS protocol policy.
+	// CoreDNS uses the protocol of the originating client request as the upstream protocol.
+	ProtocolPolicyNone ProtocolPolicy = "None"
```

Comment:
This value really indicates that the cluster admin has no preference, and we have been using the string "NoOpinion" or more often the empty string to indicate no preference.
```suggestion
	// ProtocolPolicyDefault specifies no opinion for DNS protocol.
	// Currently, this means that CoreDNS uses the protocol of the originating client request as the upstream protocol.  Note that this behavior is subject to change.  
	ProtocolPolicyDefault ProtocolPolicy = ""
```